### PR TITLE
Update packaging and CI config for CJ9

### PR DESCRIPTION
**Pull request by shenanigansd**
- Update Python version in sample packaging files and in CI to be 3.10
- Update linting tool versions [according to what bot is using](https://github.com/python-discord/bot/blob/main/pyproject.toml) 
- Remove unwanted flake8 plugins and update ignores (T000 could never be raised)
- [CI] Update action versions to the newest major versions
- [CI] Remove caching configuration since it's not doing anything useful (and we would rather not have teams deal with caching issues)

### CODE REVIEW

**Comment by ichard26** [comment]
Huh, the lint job is still running twice. I'll take a look later since I am busy right now.

**Comment by shenanigansd** [comment]
> Huh, the lint job is still running twice. I'll take a look later since I am busy right now.

Per conversation with @jb3, I went ahead and updated the `push` trigger to only run on pushes to `main`.

**Review by jb3 — APPROVED** [review]

**Review by ichard26 — APPROVED** [review]
_This review doesn't apply to the CI changes as I made them and trusting myself with a self-review feels sketchy :P_

LGTM. I tested out the new flake8 setup by installing `dev-requirements.txt` in a fresh virtual environment and then ran `flake8` against some codebases I have locally (Black and diff-shades). The output seemed reasonable.

I can also confirm pre-commit doesn't crash now. (i.e. #4 will no longer be necessary after this PR)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -18,14 +18,14 @@ repos:
     hooks:
       - id: python-check-blanket-noqa
 
-  - repo: https://github.com/pre-commit/mirrors-isort
```
Comment (ichard26) [inline .pre-commit-config.yaml:21]: FYI, https://github.com/pre-commit/mirrors-isort is deprecated hence this change.

**Review by janine9vn — APPROVED** [review]
Excellent work and I'm satisfied with the local testing done!

**Comment by ichard26** [comment]
Sorry for the messy commit history, if we want to squash merge this and use this as the message, go ahead:

```
Update packaging, linting, and CI config for CJ9

- Update Python version in sample packaging files and in CI to be 3.10.
   We want to encourage the use of the latest stable version.

- Update linting tool versions according to what bot is using
- Remove unwanted flake8 plugins and update ignores

   We decided we didn't want to enforce annotations or security audits
   (bandit) as it'd add too much friction for a week-long project that
   is supposed to be fun even for people with little experience. Also,
   T000 could never be raised as flake8-todo is not used anyway.

- [CI] Update action versions to the newest major versions
- [CI] Remove caching configuration since it's not doing anything useful
        .. and we would rather not have teams deal with caching issues.
```

**Pull request merged**